File: find_longest_substring_without_repeating_characters.js
function find_longest_substring_without_repeating_characters(inputString) {
	var conunterOfCharacter = 0;
	var temporaryLongestString = "";
	var actualLongestString = "";
	var indexOfRepeeatedChar = 0;

	while (conunterOfCharacter < inputString.length) {
		if (temporaryLongestString.includes(inputString[conunterOfCharacter])) {
			indexOfRepeeatedChar = temporaryLongestString.indexOf(inputString[conunterOfCharacter]) + 1;
			if (actualLongestString.length < temporaryLongestString.length) {
				actualLongestString = temporaryLongestString;
				temporaryLongestString = temporaryLongestString.substring(indexOfRepeeatedChar);
				temporaryLongestString += inputString[conunterOfCharacter];
			} else {
				temporaryLongestString = temporaryLongestString.substring(indexOfRepeeatedChar);
				temporaryLongestString += inputString[conunterOfCharacter];
			}
		} else {
			temporaryLongestString += inputString[conunterOfCharacter]
			if (actualLongestString.length < temporaryLongestString.length) {
				actualLongestString = temporaryLongestString;
			}
		}
		conunterOfCharacter += 1;
	}
	return actualLongestString;
}

console.log(find_longest_substring_without_repeating_characters("nasoakdnalllsandlnaiwwwnnkss"))

File: find_longest_substring_without_repeating_characters.rb
def find_longest_substring_without_repeating_characters(input_string)
	counter_of_character = 0
	temporary_longest_character = ""
	actual_longest_character = ""

	while counter_of_character < input_string.length
		if temporary_longest_character.include? input_string[counter_of_character]
			index_of_temporary_longest_string = temporary_longest_character.index(input_string[counter_of_character]) + 1
			if actual_longest_character.length < temporary_longest_character.length
				actual_longest_character = temporary_longest_character
				temporary_longest_character = temporary_longest_character[index_of_temporary_longest_string..-1]
				temporary_longest_character += input_string[counter_of_character]
			else
				temporary_longest_character = temporary_longest_character[index_of_temporary_longest_string..-1]
				temporary_longest_character += input_string[counter_of_character]
			end
		else
			temporary_longest_character += input_string[counter_of_character]
			if actual_longest_character.length < temporary_longest_character.length
				actual_longest_character = temporary_longest_character
			end
		end
		counter_of_character += 1
	end

	puts ("Longest String: " + actual_longest_character + " Length: " + actual_longest_character.length.to_s)
end

puts find_longest_substring_without_repeating_characters("nasoakdnalllsandlnaiwwwnnkss")